### ISSUE TITLE
NameError: global name 'logger' is not defined & NameError: global name 'virtual_file_system' is not defined

### ISSUE THREAD

**Issue by Spider85** [ydkhatri/UnifiedLogReader#31]
Hi,

Iam trying to use this script for some IOS Unified Logs (dont know if it would work?)
But when using i get two erros:

> Traceback (most recent call last):
>   File "UnifiedLogReader.py", line 500, in <module>
>     if not Main():
>   File "UnifiedLogReader.py", line 456, in Main
>     logger.addHandler(log_console_handler)
> NameError: global name 'logger' is not defined

>   File "UnifiedLogReader.py", line 498, in <module>
>     if not Main():
>   File "UnifiedLogReader.py", line 462, in Main
>     unified_log_reader = UnifiedLogReader()
>   File "UnifiedLogReader.py", line 286, in __init__
>     self._vfs = virtual_file_system.VirtualFileSystem(
> NameError: global name 'virtual_file_system' is not defined

Do you have an idea how to fix it?

**Comment by ydkhatri** [comment]
Are you using the code from here? Or did you do a `pip install unifiedlog`?

**Comment by ydkhatri** [comment]
Perhaps it seems that you are trying to run this as an individual script, however this only runs as a package. You will need all of the files.

**Comment by ydkhatri** [comment]
If you are trying to run from code, pull the UnifiedlogReader.py out of the scripts folder one level up, then run from there.

**Comment by Spider85** [comment]
> If you are trying to run from code, pull the UnifiedlogReader.py out of the scripts folder one level up, then run from there.

Oke that did the job! :-), but now i get only errors in this format:

```
ERROR - Exception while processing log @ 0xAE8 ct=750892553655, skipping that log entry!
Traceback (most recent call last):
  File "\UnifiedLog\tracev3_file.py", line 1004, in ProcessDataChunk
    cache_b1 = dsc_cache.GetUuidEntryFromVirtualOffset(u5)
AttributeError: 'Uuidtext' object has no attribute 'GetUuidEntryFromVirtualOffset'
```
Any idea?

**Comment by Spider85** [comment]
> If you are trying to run from code, pull the UnifiedlogReader.py out of the scripts folder one level up, then run from there.

That did the job :-)

Now i got the following when running the script with:

> py UnifiedLogReader.py .\db\diagnostics\ .\db\diagnostics\timesync\ .\db\uuidtext\ .\test\ -f SQLITE 

i Get:

```
Creating output folder .\test
INFO - Trying to create new database file at .\test\unifiedlogs.sqlite.
INFO - Started processing
ERROR -
Traceback (most recent call last):
  File "\UnifiedLog\Lib.py", line 150, in ParseFolder
    entries = self.vfs.listdir(dsc_path)
  File "\UnifiedLog\virtual_file_system.py", line 26, in listdir
    return os.listdir(path)
WindowsError: [Error 3] Het systeem kan het opgegeven pad niet vinden: u'.\\db\\diagnostics\\dsc\\*.*'
INFO - Finished in time = 00:00:00
INFO - 0 Logs processed
INFO - Review the Log file and report any ERRORs or EXCEPTIONS to the developers
```
Any idea?

**Comment by ydkhatri** [comment]
Well I can't read Dutch, but google translate tells me `Het systeem kan het opgegeven pad niet vinden` means `The system cannot find the specified path`
It is expecting a `dsc` folder in the diagnostics folder. The script cannot proceed without it.

**Comment by ydkhatri** [comment]
OK, I now see that you are trying to parse IOS logs, it may be a different folder there. I have not looked at IOS.

**Comment by Spider85** [comment]
> OK, I now see that you are trying to parse IOS logs, it may be a different folder there. I have not looked at IOS. But I will be next week, and will get back to you then.

Superb Tnx i will wait on it !

**Comment by ydkhatri** [comment]
Can you try running it like this:
`py UnifiedLogReader.py .\db\uuidtext\ .\db\diagnostics\timesync\ .\db\diagnostics\ .\test\ -f SQLITE`

The order of the arguments is: uuidtext, timesync, tracev3(in diagnostics)

**Comment by ydkhatri** [comment]
I made some bug fixes, and tested with **iOS 12.4** logs now. Seems to work fine. I am closing this bug, feel free to reopen if you have issues.
